lib/model.js

```diff
@@ -1466,10 +1466,8 @@ Model.prototype.findById = function(param, options) {
     throw new Error('Argument passed to findById is invalid: '+param);
   }
 
-  // Bypass a possible overloaded findAll.
-  return Model.prototype.findAll.call(this, _.defaults(options, {
-    plain: true
-  }));
+  // Bypass a possible overloaded findOne
+  return Model.prototype.findOne.call(this, options);
```

Comment:
@mickhansen you are correct, I hadn't noticed [your commit](https://github.com/treythomas123/sequelize/commit/35b9558fc09b75239b6cab955dba9546a8ec9507) that removed the limit on `findById`.  I had been using a slightly older version where it was still adding the limit, and made this change without checking to see if it was still broken on master. 

I think `findOne` still makes sense here though -- it's a little cleaner since the addition `{plain: true}` option doesn't have to be repeated (it's already being added within `findOne`).